File: seed/csharp-model/trace/src/SeedTraceClient/Submission/SubmissionStatusV2.cs
using SeedTraceClient;
using System.Text.Json.Serialization;

namespace SeedTraceClient;

public class SubmissionStatusV2
{
    public class _Test : TestSubmissionStatusV2
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "test";
    }
    public class _Workspace : WorkspaceSubmissionStatusV2
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "workspace";
    }
}

File: seed/ruby-sdk/trace/lib/seed/submission/types/submission_status_v_2.rb
# frozen_string_literal: true

module Seed
  module Submission
    module Types
      class SubmissionStatusV2 < Internal::Types::Model
        extend Seed::Internal::Types::Union

        discriminant :type

        member -> { Seed::Submission::Types::TestSubmissionStatusV2 }, key: "TEST"
        member -> { Seed::Submission::Types::WorkspaceSubmissionStatusV2 }, key: "WORKSPACE"
      end
    end
  end
end
